File: Data Structures & Algorithms/largest-unique-number/submission-0.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        count = {}
        for num in nums:
            count[num] = count.get(num,0) + 1
            result = -1
        for num in nums:
            if count[num] == 1 and num > result:
                result = num
        return result
            

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java



class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        // 1. Count
        Map<Integer, Integer> count = new HashMap<>();
        for (int n : nums) {
            count.put(n, count.getOrDefault(n, 0) + 1);
        }

        // 2. Dump entries into a list and sort by count, highest first
        List<Map.Entry<Integer, Integer>> entries = new ArrayList<>(count.entrySet());
        entries.sort((a, b) -> b.getValue() - a.getValue());

        // 3. Take the first k keys
        int[] res = new int[k];
        for (int i = 0; i < k; i++) {
            res[i] = entries.get(i).getKey();
        }
        return res;
    }
}